lib/classes/Variables.js

```diff
@@ -333,6 +340,14 @@ class Variables {
 
         return output.OutputValue;
       });
+    if (!this.cfVars) {
+      this.cfVars = {};
+    }
+    if (!this.cfVars[stackName]) {
+      this.cfVars[stackName] = {};
+    }
+    this.cfVars[stackName][outputLogicalId] = promise;
```

Comment:
This whole if/if can be turned into a single line with lodash:
```
_.set(cfVars, `${stackName}${outputLogicalId}`, promise);
```
This will automatically create all needed path objects if they do not exist.

Reply:
Thanks for that. You may have guessed I am very new to javascript :)